File: ServerSpace/src/main/java/DataLayer/ServListaDB.java
package DataLayer;


import EntityLayer.ServListaEntity;
import Enumerados.ProcessActionEnum;
import Framework.BaseDB;
import Framework.Inj;
import Framework.Utilidades;
import Framework.injector;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

public class ServListaDB extends BaseDB {
    public ArrayList<ServListaEntity> ObtenerCodigo(String codigo) {

        injector InjDB = new injector();
        ArrayList<ServListaEntity> DatoMemoria = new ArrayList<>();
        ServListaEntity en;
        try {
            InjDB.Sp("sp_ServListaObtenerCodigo");
            InjDB.Pmt_String("@Codigo", codigo, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new ServListaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }


    public ArrayList<ServListaEntity> Buscar(String codigo, String Nombre){

        injector InjDB = new injector();
        ArrayList<ServListaEntity> DatoMemoria = new ArrayList<>();
        ServListaEntity en;
        try {
            InjDB.Sp("sp_ServListaBuscar");
            InjDB.Pmt_String("@Codigo", codigo, false);
            InjDB.Pmt_String("@Nombre", Nombre, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new ServListaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public ArrayList<ServListaEntity> ObtenerMain(String codigo) {

        injector InjDB = new injector();
        ArrayList<ServListaEntity> DatoMemoria = new ArrayList<>();
        ServListaEntity en;
        try {
            InjDB.Sp("sp_ServListaObtenerMain");
            InjDB.Pmt_String("@Codigo", codigo, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new ServListaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }


    
    public ServListaEntity Registrar(ServListaEntity entity) {
        Boolean State = null;
        try {
            Inj.IniciarTranssaccion(false);
            String Store = "sp_ServListaRegistrar";
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Store = "sp_ServListaActualizar";
            }
            Inj.Sp(Store);
            Inj.Pmt_Integer("@ListaId", entity.getListaId(), true);
            Inj.Pmt_Integer("@CampoId", entity.getCampoId(), false);
            Inj.Pmt_String("@Codigo", entity.getCodigo(), false);
            Inj.Pmt_String("@Nombre", entity.getNombre(), false);
            Inj.Pmt_String("@Descripcion", entity.getDescripcion(), false);
            Inj.Pmt_String("@CodUsuario", entity.getCodUsuario(), false);
            Inj.Pmt_String("@FechaRegistro", Utilidades.getFechaRegistro(), false);
            Inj.Pmt_Boolean("@EstadoRegistro", entity.getEstadoRegistro(), false);
            if (entity.getAction() == ProcessActionEnum.Add.getValor()) {
                int Id = Inj.RunInsert();
                State = Id > 0;
                if (State) {
                    entity.setListaId(Id);
                }
            }
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Inj.RunUpdate();
            }

        } catch (Exception ex) {
            throw new UnsupportedOperationException("Datalater : " + ex);
        }
        return entity;
    }


    public ArrayList<ServListaEntity> ObtenerItem(int ListaId) {

        injector InjDB = new injector();
        ArrayList<ServListaEntity> DatoMemoria = new ArrayList<>();
        ServListaEntity en;
        try {
            InjDB.Sp("sp_ServListaObtenerItem");
            InjDB.Pmt_Integer("@ListaId", ListaId, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new ServListaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }
}

File: Space/Space.DataLayer/ServListaDB.cs
using Framework;
using Space.EntityLayer;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Space.DataLayer
{
    public class ServListaDB : BaseDB
    {
        public List<ServListaEntity> ObtenerServListaCodigo(String codigo)
        {
            Conexion.OpenConnection();
            Conexion.Store("sp_ServListaObtenerCodigo");
            Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, codigo);

            SqlDataReader reader = Conexion.ReaderData();
            List<ServListaEntity> ListaData = new List<ServListaEntity>();
            FillSchemeTable(reader);
            while (reader.Read())
            {
                ServListaEntity entity = new ServListaEntity();
                if (FillFrom<ServListaEntity>(reader, entity)) ListaData.Add(entity);
            }
            reader.Close();
            Conexion.CloseConnection();
            return ListaData;
        }
        public List<ServListaEntity> BuscarServLista(String codigo, String Nombre)
        {
            Conexion.OpenConnection();
            Conexion.Store("sp_ServListaBuscar");
            Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, codigo);
            Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 100, ParameterDirection.Input, Nombre);
            SqlDataReader reader = Conexion.ReaderData();
            List<ServListaEntity> ListaData = new List<ServListaEntity>();
            FillSchemeTable(reader);
            while (reader.Read())
            {
                ServListaEntity entity = new ServListaEntity();
                if (FillFrom<ServListaEntity>(reader, entity)) ListaData.Add(entity);
            }
            reader.Close();
            return ListaData;
        }

        public List<ServListaEntity> ObtenerServListaMain(String codigo)
        {
            Conexion.OpenConnection();
            Conexion.Store("sp_ServListaObtenerMain");
            Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, codigo);

            SqlDataReader reader = Conexion.ReaderData();
            List<ServListaEntity> ListaData = new List<ServListaEntity>();
            FillSchemeTable(reader);
            while (reader.Read())
            {
                ServListaEntity entity = new ServListaEntity();
                if (FillFrom<ServListaEntity>(reader, entity)) ListaData.Add(entity);
            }
            reader.Close();
            Conexion.CloseConnection();
            return ListaData;
        }
        public Int32 Registrar(ServListaEntity Ent)
        {
            try
            {

                String NameStore = "sp_ServListaRegistrar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("ListaId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.ListaId);
                Conexion.AddParemeter("CodigoCampo", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodigoCampo);
                Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Codigo);
                Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Nombre);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);
                Ent.ListaId = Conexion.ExecuteQueryItem("ListaId");

                return Ent.ListaId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }

        }



        public Int32 Actualizar(ServListaEntity Ent)
        {
            try
            {

                String NameStore = "sp_ServListaActualizar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("ListaId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.ListaId);
                Conexion.AddParemeter("CodigoCampo", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodigoCampo);
                Conexion.AddParemeter("Codigo", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Codigo);
                Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 100, ParameterDirection.Input, Ent.Nombre);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);
                Ent.ListaId = Conexion.ExecuteQueryItem("ListaId");

                return Ent.ListaId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }

        }
        public List<ServListaEntity> ObtenerServListaItem(Int32 ListaId)
        {
            Conexion.OpenConnection();
            Conexion.Store("sp_ServListaObtenerItem");
            Conexion.AddParemeter("ListaId", SqlDbType.Int, 20, ParameterDirection.Input, ListaId);

            SqlDataReader reader = Conexion.ReaderData();
            List<ServListaEntity> ListaData = new List<ServListaEntity>();
            FillSchemeTable(reader);
            while (reader.Read())
            {
                ServListaEntity entity = new ServListaEntity();
                if (FillFrom<ServListaEntity>(reader, entity)) ListaData.Add(entity);
            }
            reader.Close();
            Conexion.CloseConnection();
            return ListaData;
        }
    }
}
